MediaRecorder.bs

```diff
@@ -196,23 +216,44 @@ interface MediaRecorder : EventTarget {
     <li>If {{state}} is not {{inactive}}, throw an {{InvalidStateError}}
     {{DOMException}} and abort these steps.</li>
 
-    <li>If the {{MediaRecorder/stream}}'s <a>isolation properties</a> disallow access
+    <li>If the {{stream}}'s <a>isolation properties</a> disallow access
     from this {{MediaRecorder}}, throw a {{SecurityError}} {{DOMException}} and
     abort these steps.</li>
 
+    <li>If the {{stream}} is {{inactive}}, throw a {{NotSupportedError}}
+    {{DOMException}} and abort these steps.</li>
+
+    <li>Let |tracks| be the set of {{live}} tracks in the {{stream}}'s
+    [=track set=].</li>
+
+    <li>If the [=constrainedMimeType=] internal property specifies a media type,
+    container, or codec such that the User Agent cannot record a track in
+    |tracks| with it, throw a {{NotSupportedError}} {{DOMException}} and abort
+    these steps.</li>
+
+    <li>Set {{mimeType}} to the [=constrainedMimeType=] internal property,
+    extended by the media type, subtype and codecs parameter reflecting the
+    configuration used by the MediaRecorder to record all tracks in |tracks|.
+    The media type, subtype and any parameters in the [=constrainedMimeType=]
+    internal property, MUST NOT be changed. {{mimeType}} MAY be extended with
+    the profiles parameter [[RFC6381]] or further codec-specific parameters to
+    avoid ambiguity.</li>
+
     <li>Set {{state}} to {{recording}}, and run the following steps in parallel:
     <ol>
-      <li>Start gathering the data into a {{Blob}} <var>blob</var> and queue a
-      task, using the DOM manipulation task source, to <a>fire an event</a>
-      named <a>start</a> at <var>target</var>, then return <code>undefined</code>.
-      </li>
-
-      <li>If at any point the {{MediaRecorder/stream}}'s <a>isolation properties</a>
-      change so that {{MediaRecorder}} is no longer allowed access to it, the
-      UA MUST immediately stop gathering data, discard any data that it has
-      gathered, and queue a task, using the DOM manipulation task source, that
-      runs the following steps:
+      <li>Start recording all tracks in |tracks| and gather the data into a
```

Comment:
Maybe "start recording all tracks in |tracks| using the MediaRecorder's current configuration, and gather..."?